### PR TITLE
Support resolving variables when calling `dump`

### PR DESCRIPTION
**Pull request by mhahn**
We now resolve variables when calling `dump`.

We resolve the variables with a `provider` set to`None` to prevent calls to the provider. `dump` is analogous to "dry run without building anything", so there should be no need to access the provider, run `pre_build` or `post_build` hooks.

### CODE REVIEW

**Comment by mhahn** [comment]
Depends on merging: https://github.com/remind101/stacker/pull/229

File: stacker/actions/build.py [hunk 1]
```diff
@@ -253,10 +253,10 @@ def _get_dependencies(self):
             dependencies[stack.fqn] = stack.requires
         return dependencies
 
-    def pre_run(self, outline=False, *args, **kwargs):
+    def pre_run(self, outline=False, dump=False, *args, **kwargs):
         """Any steps that need to be taken prior to running the action."""
         pre_build = self.context.config.get("pre_build")
-        if not outline and pre_build:
+        if not outline and not dump and pre_build:
```
Comment (phobologic) [inline stacker/actions/build.py]: This logic is getting a little long, should we maybe make this a method? `should_run_hooks` or something?

File: stacker/lookups/registry.py [hunk 2]
```diff
@@ -24,6 +24,19 @@ def register_lookup_handler(lookup_type, handler_or_path):
     LOOKUP_HANDLERS[lookup_type] = handler
 
 
+def unregister_lookup_handler(lookup_type):
```
Comment (phobologic) [inline stacker/lookups/registry.py]: Is this only used in tests?

Comment (mhahn) [inline stacker/lookups/registry.py]: yes

Comment (mhahn) [inline stacker/lookups/registry.py]: the docstring:

```    """Unregister the specified lookup type.

    This is useful when testing various lookup types if you want to unregister
    the lookup type after the test runs.

    Args:
        lookup_type (str): Name of the lookup type to unregister

    """
```

**Pull request merged**